`mypy` issues

Something has been released recently upstream, triggering `mypy` to fail with:
```
webviz_subsurface/_providers/ensemble_summary_provider/_resampling.py:87: error: Need type annotation for "padded_y"  [var-annotated]
webviz_subsurface/_providers/ensemble_summary_provider/_resampling.py:87: error: Argument 1 to "concatenate" has incompatible type "Tuple[ndarray[Any, Any], List[float]]"; expected "Union[_SupportsArray[dtype[<nothing>]], Sequence[_SupportsArray[dtype[<nothing>]]], Sequence[Sequence[_SupportsArray[dtype[<nothing>]]]], Sequence[Sequence[Sequence[_SupportsArray[dtype[<nothing>]]]]], Sequence[Sequence[Sequence[Sequence[_SupportsArray[dtype[<nothing>]]]]]]]"  [arg-type]
webviz_subsurface/plugins/_volumetric_analysis/controllers/comparison_controllers.py:319: error: Incompatible return value type (got "ndarray[Any, dtype[Any]]", expected "List[Any]")  [return-value]
webviz_subsurface/plugins/_line_plotter_fmu/controllers/build_figure.py:250: error: Incompatible return value type (got "floating[Any]", expected "List[float]")  [return-value]
webviz_subsurface/plugins/_line_plotter_fmu/controllers/build_figure.py:253: error: Incompatible return value type (got "floating[Any]", expected "List[float]")  [return-value]
webviz_subsurface/plugins/_simulation_time_series/utils/vector_statistics.py:32: error: Incompatible return value type (got "floating[Any]", expected "List[float]")  [return-value]
webviz_subsurface/plugins/_simulation_time_series/utils/vector_statistics.py:35: error: Incompatible return value type (got "floating[Any]", expected "List[float]")  [return-value]
webviz_subsurface/plugins/_simulation_time_series/utils/vector_statistics.py:38: error: Incompatible return value type (got "floating[Any]", expected "List[float]")  [return-value]
webviz_subsurface/_utils/simulation_timeseries.py:58: error: Incompatible return value type (got "floating[Any]", expected "List[float]")  [return-value]
webviz_subsurface/_utils/simulation_timeseries.py:61: error: Incompatible return value type (got "floating[Any]", expected "List[float]")  [return-value]
webviz_subsurface/_utils/simulation_timeseries.py:64: error: Incompatible return value type (got "floating[Any]", expected "List[float]")  [return-value]
```

--- a/webviz_subsurface/_providers/ensemble_summary_provider/_resampling.py
+++ b/webviz_subsurface/_providers/ensemble_summary_provider/_resampling.py
@@ -84,7 +84,7 @@ def interpolate_backfill(
     # Finds the leftmost valid insertion indices for the values x in xp
     indices = np.searchsorted(xp, x, side="left")
 
-    padded_y = np.concatenate((yp, [yright]))
+    padded_y = np.concatenate((yp, np.array(yright)))
 
     ret_arr = padded_y[indices]
 

--- a/webviz_subsurface/_providers/ensemble_summary_provider/_resampling.py
+++ b/webviz_subsurface/_providers/ensemble_summary_provider/_resampling.py
@@ -84,7 +84,7 @@ def interpolate_backfill(
     # Finds the leftmost valid insertion indices for the values x in xp
     indices = np.searchsorted(xp, x, side="left")
 
-    padded_y = np.concatenate((yp, np.array(yright)))
+    padded_y = np.concatenate((yp, np.array([yright])))
 
     ret_arr = padded_y[indices]
 

--- a/webviz_subsurface/_utils/simulation_timeseries.py
+++ b/webviz_subsurface/_utils/simulation_timeseries.py
@@ -54,13 +54,13 @@ def calc_series_statistics(
     refaxis is used if another column than DATE should be used to groupby.
     """
     # Invert p10 and p90 due to oil industry convention.
-    def p10(x: List[float]) -> List[float]:
+    def p10(x: List[float]) -> np.floating:
         return np.nanpercentile(x, q=90)
 
-    def p90(x: List[float]) -> List[float]:
+    def p90(x: List[float]) -> np.floating:
         return np.nanpercentile(x, q=10)
 
-    def p50(x: List[float]) -> List[float]:
+    def p50(x: List[float]) -> np.floating:
         return np.nanpercentile(x, q=50)
 
     # Calculate statistics, ignoring NaNs.

--- a/webviz_subsurface/plugins/_line_plotter_fmu/controllers/build_figure.py
+++ b/webviz_subsurface/plugins/_line_plotter_fmu/controllers/build_figure.py
@@ -246,10 +246,10 @@ def calc_series_statistics(
     refaxis is used if another column than DATE should be used to groupby.
     """
     # Invert p10 and p90 due to oil industry convention.
-    def p10(x: List[float]) -> List[float]:
+    def p10(x: List[float]) -> np.floating:
         return np.nanpercentile(x, q=90)
 
-    def p90(x: List[float]) -> List[float]:
+    def p90(x: List[float]) -> np.floating:
         return np.nanpercentile(x, q=10)
 
     # Calculate statistics, ignoring NaNs.

--- a/webviz_subsurface/plugins/_simulation_time_series/utils/vector_statistics.py
+++ b/webviz_subsurface/plugins/_simulation_time_series/utils/vector_statistics.py
@@ -28,13 +28,13 @@ def create_vectors_statistics_df(vectors_df: pd.DataFrame) -> pd.DataFrame:
     )
 
     # Invert p10 and p90 due to oil industry convention.
-    def p10(x: List[float]) -> List[float]:
+    def p10(x: List[float]) -> np.floating:
         return np.nanpercentile(x, q=90)
 
-    def p90(x: List[float]) -> List[float]:
+    def p90(x: List[float]) -> np.floating:
         return np.nanpercentile(x, q=10)
 
-    def p50(x: List[float]) -> List[float]:
+    def p50(x: List[float]) -> np.floating:
         return np.nanpercentile(x, q=50)
 
     statistics_df: pd.DataFrame = (

--- a/webviz_subsurface/plugins/_volumetric_analysis/controllers/comparison_controllers.py
+++ b/webviz_subsurface/plugins/_volumetric_analysis/controllers/comparison_controllers.py
@@ -312,7 +312,7 @@ def create_comparison_df(
 
 def compute_highlighted_col(
     df: pd.DataFrame, response: str, value1: str, selections: dict
-) -> list:
+) -> np.ndarray:
     highlight_mask = (df[response][value1] > selections["Ignore <"]) & (
         df[response]["diff (%)"].abs() > selections["Accept value"]
     )

